.github/scripts/generate_binary_build_matrix.py

```diff
@@ -24,7 +24,21 @@
 
 CPU_AARCH64_ARCH = ["cpu-aarch64"]
 
-PYTORCH_EXTRA_INSTALL_REQUIREMENTS = (
+PYTORCH_EXTRA_INSTALL_REQUIREMENTS_CU118 = (
```

Comment:
@ptrblck Perhaps we can create dictionary instead of two strings
```
PYTORCH_EXTRA_INSTALL_REQUIREMENTS = {
"11.8": (...),
"12.1" :  (
    "nvidia-cuda-nvrtc-cu12==12.1.105; platform_system == 'Linux' and platform_machine == 'x86_64' | "  # noqa: B950
    "nvidia-cuda-runtime-cu12==12.1.105; platform_system == 'Linux' and platform_machine == 'x86_64' | "
    "nvidia-cuda-cupti-cu12==12.1.105; platform_system == 'Linux' and platform_machine == 'x86_64' | "
    "nvidia-cudnn-cu12==8.9.2.26; platform_system == 'Linux' and platform_machine == 'x86_64' | "
    "nvidia-cublas-cu12==12.1.3.1; platform_system == 'Linux' and platform_machine == 'x86_64' | "
    "nvidia-cufft-cu12==11.0.2.54; platform_system == 'Linux' and platform_machine == 'x86_64' | "
    "nvidia-curand-cu12==10.3.2.106; platform_system == 'Linux' and platform_machine == 'x86_64' | "
    "nvidia-cusolver-cu12==11.4.5.107; platform_system == 'Linux' and platform_machine == 'x86_64' | "
    "nvidia-cusparse-cu12==12.1.0.106; platform_system == 'Linux' and platform_machine == 'x86_64' | "
    "nvidia-nccl-cu12==2.19.3; platform_system == 'Linux' and platform_machine == 'x86_64' | "
    "nvidia-nvtx-cu12==12.1.105; platform_system == 'Linux' and platform_machine == 'x86_64'"
)
}
```
Then we can use it like
```
PYTORCH_EXTRA_INSTALL_REQUIREMENTS[arch_version]
```

Reply:
I don't think it matters at this point, but yes, would be nice to have Linux+x86 constraint defined only once